src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1561,6 +1752,45 @@ LogicalResult ONNXUnsqueezeOp::inferShapes() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// Cast
+
+LogicalResult ONNXCastOp::inferShapes() {
```

Comment:
Our apologies, it seems like cast op has already been supported in a very recent patch (https://github.com/onnx/onnx-mlir/pull/156/files).

@tungld : wonder if there's anything we should add to the CastOp from here?

Reply:
We have a helper function to convert ONNX type to MLIR type, please use it to make type conversion consistent. You can replace `switch-case` by something similar to:
```mlir
if (auto elementType = convertONNXTypeToMLIRType(builder, static_cast<onnx::TensorProto_DataType>(targetType)))
  getResult().setType(getOutputType(elementType));
else
  return emitError(...);
```

Reply:
@tjingrant that patch just gives an unranked tensor unless I'm mistaken? I don't think there's enough information at that stage to do shape inference yet, so it makes sense.